### DISCUSSION TITLE
Incremental Hash Calculation in OpenSSL 3.0.1 and Later

### DISCUSSION THREAD

**Discussion by tkconsulting** [openssl/openssl#25251] [Q&A]
In OpenSSL version 3.0.1 and later, the function calls EVP_MD_CTX_get0_md_data() (also known as EVP_MD_CTX_md_data()) return NULL when attempting to retrieve the digest state for serialization and future restoration to incrementally calculate a hash.

How can we achieve incremental hash calculation without leveraging md_data?
Are there any flags or compilation parameters that would enable access to md_data?
Can anyone point to the release notes or documentation where the change to the returned value (NULL) of the function EVP_MD_CTX_get0_md_data() is described?

**Comment by kulkarniamit** [comment]
Related issue: https://github.com/openssl/openssl/issues/14222
Related discussion: https://github.com/openssl/openssl/discussions/24501
